packages/supi/src/install/extendInstallOptions.ts

```diff
@@ -174,6 +174,14 @@ export default async (
   if (extendedOpts.userAgent.startsWith('npm/')) {
     extendedOpts.userAgent = `${extendedOpts.packageManager.name}/${extendedOpts.packageManager.version} ${extendedOpts.userAgent}`
   }
+  if (extendedOpts.lockfileOnly && !extendedOpts.ignoreScripts) {
+    if (opts.ignoreScripts === false) {
+      throw new PnpmError('CONFIG_CONFLICT_LOCKFILE_ONLY_WITH_NO_IGNORE_SCRIPTS',
+        'Cannot execute the scripts because lockfileOnly is set to false')
+    } else {
+      extendedOpts.ignoreScripts = true
+    }
+  }
```

Comment:
Seems like it will have to be like this, otherwise the tests will fail.

```suggestion
  if (extendedOpts.lockfileOnly) {
    extendedOpts.ignoreScripts = true
  }
```